### ISSUE TITLE
Installer issue with composer - Function not found

### ISSUE THREAD

**Issue by steve-allan** [akeneo/pim-community-dev#7185]
I've tried to install the latest available version of Akeneo PIM community on Ubuntu 16 with PHP 7.1
I've follwed the instructions here:
https://github.com/akeneo/pim-community-standard#using-composer-to-create-the-project

and here:
https://docs.akeneo.com/2.0/install_pim/installation_ce_archive.html

At the instruction:
php bin/console cache:clear --no-warmup --env=prod

I get the following error:

PHP Fatal error: Uncaught Symfony\Component\Debug\Exception\UndefinedFunctionException: Attempted to call function "apcu_exists" from namespace "Doctrine\Common\Cache". in /home/steve/pim-project/vendor/doctrine/cache/lib/Doctrine/Common/Cache/ApcuCache.php:44

I've looked at that file and no apcu_exists method is in there.

What step have I done wrong?
Thanks,
Steve

**Comment by damien-carcel** [comment]
Hello @steve-allan .

It seems APCU is not installed or not activated.
Try to run `php -i | grep apc`. You should have something like that:
```
Additional .ini files parsed => /etc/php/conf.d/apcu.ini,
apcu
apc.coredump_unmap => Off => Off
apc.enable_cli => On => On
apc.enabled => On => On
apc.entries_hint => 4096 => 4096
apc.gc_ttl => 3600 => 3600
apc.mmap_file_mask => no value => no value
apc.preload_path => no value => no value
apc.serializer => php => php
apc.shm_segments => 1 => 1
apc.shm_size => 32M => 32M
apc.slam_defense => On => On
apc.smart => 0 => 0
apc.ttl => 0 => 0
apc.use_request_time => On => On
apc.writable => /tmp => /tmp
```
Be sure that `apc.enable_cli` and `apc.enabled` are `On`.

Regards,
Damien

**Comment by Agost15** [comment]
Hi, thanks for the solution but I would like to know if there is not an error. The error in APCU and the solution is for APC, is it correct?
I need the solution to this. This is my mistake
Fatal error: Uncaught Symfony\Component\Debug\Exception\UndefinedFunctionException: Attempted to call function "apcu_exists" from namespace "Doctrine\Common\Cache". in ..\akeneo\vendor\doctrine\cache\lib\Doctrine\Common\Cache\ApcuCache.php:44 Stack trace: #0 ..\akeneo\vendor\doctrine\cache\lib\Doctrine\Common\Cache\ChainCache.php(62): Doctrine\Common\Cache\ApcuCache->doContains('DoctrineNamespa...') #1 ..\akeneo\vendor\doctrine\cache\lib\Doctrine\Common\Cache\CacheProvider.php(212): Doctrine\Common\Cache\ChainCache->doFetch('DoctrineNamespa...') #2 ..\akeneo\vendor\doctrine\cache\lib\Doctrine\Common\Cache\CacheProvider.php(185): Doctrine\Common\Cache\CacheProvider->getNamespaceVersion() #3 ..\akeneo\vendor\doctrine\cache\lib\Doctrine\Common\Cache\CacheProvider.php(78): Doctrine\Common\Cache\CacheProvider->getNamespacedId('Pim\\Bundle\\Cata...') #4 ..\akeneo\vendor\doctrine\common\lib\Doctrine\Common\Persistence\Mapping\AbstractClassMetadataFactory.php(211): in ..\akeneo\vendor\doctrine\cache\lib\Doctrine\Common\Cache\ApcuCache.php on line 44

**Comment by damien-carcel** [comment]
Hi @Agost15 .

The options are called `apc.*`, but they are for `apcu` => https://secure.php.net/manual/en/apcu.configuration.php. So your problem and the solution I proposed are for `apcu`.

From the error you past in your last comment, It seems, like for @steve-allan, that the `apcu` extension is not activated or not installed. You need to install it and/or activate it.

Regards,
Damien.

**Comment by SebastianKieper** [comment]
i have exactly the same issue. but apcu is enabled. 
This is what my php.ini says:
/etc/php/7.1/cli/conf.d/20-apcu.ini,
apcu
apc.coredump_unmap => Off => Off
apc.enable_cli => On => On
apc.enabled => On => On
apc.entries_hint => 4096 => 4096
apc.gc_ttl => 3600 => 3600
apc.mmap_file_mask => no value => no value
apc.preload_path => no value => no value
apc.serializer => php => php
apc.shm_segments => 1 => 1
apc.shm_size => 32M => 32M
apc.slam_defense => Off => Off
apc.smart => 0 => 0
apc.ttl => 0 => 0
apc.use_request_time => On => On

but i still cant clear caches and have 8 to 10 seconds loading time for each request. I have this issue, while i am on the docker containers coming with the community edition

**Comment by magnusjjj** [comment]
Why was this closed? :)

**Comment by jmleroux** [comment]
Hello @magnusjjj ,
I closed it because the installation is now based on our Docker image which is installed with APCu support by default.
Do you still have the same issue with the Docker image?

**Comment by magnusjjj** [comment]
Not really. Did have the issue outside of docker (our installation is an old one we have needed to upgrade). In our case, apc *was* enabled, but we needed to install the pecl package as well, which wasn't exactly clear.

I wonder, will you write documentation for how to move from a non-docker environment in the future? The documentation is.. uh. Wrong in several places and has issues that won't let it be installed on the 'supported distros' when upgraded.
